### PR TITLE
Streamline builds in preparation for Python 3.7

### PR DESCRIPTION
**Pull request by Lnaden**
Start builds against Python 3.7 for testing

### CODE REVIEW

File: conda-build-all [hunk 1]
```diff
@@ -95,14 +95,14 @@ def main():
...
-        default='27,35,36',
+        default='27,35,36,37',
 
     )
     p.add_argument(
         '--numpy',
         help="Set the NumPy version used by conda build",
         metavar="NUMPY_VER",
-        default='1.10,1.11,1.12',
+        default='1.13,1.14,1.15',
```
Comment (jchodera) [inline conda-build-all:105]: @Lnaden : It looks like conda-forge is actively building numpy 1.9, 1.11, and 1.15 for Python 3.7 now:
https://anaconda.org/conda-forge/numpy/files

They're have also built numpy 1.10, 1.12, 1.13, and 1.14 for Python 2.7, 3.5, and 3.6.

This is a pretty weird combination.

**Comment by jchodera** [comment]
@Lnaden : I've tried to update the build infrastructure to build against the [numpy versions available for each Python version on conda-forge](https://anaconda.org/conda-forge/numpy/files).

**Comment by Lnaden** [comment]
Thats fine. I tried to make as minimal changes as we need to. Honestly, we could probably (and should) remove that bit since none of our packages use the `numpy x.x` formalism anymore. We could just remove that check. They generalized that system in Conda Build 3, but that will take re-designing that script and most the recipes. You could just build against the max version (1.15) right now.

**Comment by jchodera** [comment]
Good point!

**Comment by jchodera** [comment]
Looks like conda-forge hasn't started building some of our dependencies for Python 3.7 yet, so this might have to hold for a bit:
```
Can't build /io/openmm due to unsatisfiable dependencies:
The following specifications were found to be in conflict:
  - python 3.7*
  - sphinx ==1.5.6 -> python 2.7* -> openssl 1.0.1*
  - sphinx ==1.5.6 -> python 2.7* -> readline 6.2*
  - sphinx ==1.5.6 -> python 2.7* -> tk 8.5*
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by Lnaden** [comment]
I think you can safely unpin from sphinx 1.5.6 per the PR I have [here on the main conda-recipes](https://github.com/omnia-md/conda-recipes/pull/909) channel. Just make sure its at least that to avoid other odd backwards compatibility issues

**Comment by jchodera** [comment]
@lnaden: For reference, unpinning `sphinx` doesn't seem to work:
```

The LaTeX files are in ../../build/sphinx-docs/developerguide/latex.
Run 'make' in that directory to run these through (pdf)latex
(use `make latexpdf' here to do that automatically).
Running LaTeX files through pdflatex...
gmake[5]: latexmk: Command not found
gmake[5]: *** [OpenMMDeveloperGuide.pdf] Error 127
gmake[4]: *** [latexpdf] Error 2
make[3]: *** [sphinx-docs/developerguide/latex/OpenMMDeveloperGuide.pdf] Error 2
make[2]: *** [docs-source/CMakeFiles/sphinxpdf.dir/all] Error 2
make[1]: *** [docs-source/CMakeFiles/sphinxpdf.dir/rule] Error 2
make: *** [sphinxpdf] Error 2
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 342, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 333, in execute
    noverify=args.no_verify)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 97, in build
    need_source_download=need_source_download, config=config)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1524, in build_tree
    config=config)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1147, in build
    utils.check_call_env(cmd, env=env, cwd=src_dir)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 628, in check_call_env
    return _func_defaulting_env_to_os_environ(subprocess.check_call, *popenargs, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 624, in _func_defaulting_env_to_os_environ
    return func(_args, **kwargs)
  File "/opt/conda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-e', '/opt/conda/conda-bld/openmm_1535835361528/work/conda_build.sh']' returned non-zero exit status 2.
```

**Comment by jchodera** [comment]
The  problem now is that there appear to be no `conda-forge` `sphinx` builds for `py37`.

**Pull request merged**